lib/util/Components.js

```diff
@@ -173,35 +173,41 @@ class Components {
     const usedPropTypes = {};
 
     // Find props used in components for which we are not confident
-    Object.keys(thisList).filter((i) => thisList[i].confidence < 2).forEach((i) => {
-      let component = null;
-      let node = null;
-      node = thisList[i].node;
-      while (!component && node.parent) {
-        node = node.parent;
-        // Stop moving up if we reach a decorator
-        if (node.type === 'Decorator') {
-          break;
+    for (const val of values(thisList)) {
```

Comment:
i am very skeptical that `for..of` is ever _faster_ than alternatives, and i have a very strong preference to avoid loops. I'd really prefer this to remain a filter + forEach, unless you can persuade me that the clarity loss is worth the speedup..

Reply:
On a large project with 500 files, from some very quick checking, the `list` function runs around 23,000 times so my intuition says it's worth looking at, but I'll do some specific benchmarking of this change to check.